services/server/config/config_test.go

```diff
@@ -41,7 +41,7 @@ func TestMergeConfigs(t *testing.T) {
 
 	b := &Config{
 		Root:             "new_root",
-		RequiredPlugins:  []string{"new_plugin1", "new_plugin2"},
+		RequiredPlugins:  []string{"io.containerd.new_plugin1.v1", "io.containerd.new_plugin2.v1"},
```

Comment:
Needs `Version: 2`

Reply:
Hm.. good one; looks like `Version` was used here to verify the "integer" cases; https://github.com/containerd/containerd/blob/39259a8f35919a0d02c9ecc2871ddd6ccf6a7c6e/services/server/config/config.go#L322-L323

But a config without Version was deprecated, so let me add a test case using `OomScore` for this.

We should probably look what we want the behaviour to be if the override file does not specify a version, or if the version is different from the original one; i.e., do we want `mergeConfig()` to produce an error when merging a `v2` config with a `v1` config (or `v3` with `v2`)?

Reply:
Updated; moved these changes to a separate commit as well (to separate it from the strings.Cut changes)